tests/unit/coroutines_test.cc

```diff
@@ -768,6 +774,37 @@ seastar::future<> test_async_generator_drained() {
     BOOST_REQUIRE(!sentinel.has_value());
 }
 
+template<template<typename> class Container>
```

Comment:
The entire new section is missing some verbal explanations about the tests, the expected results and how they are achieved.